src/OnChain.ts

```diff
@@ -136,62 +138,100 @@ export const OnChainMixin = (superclass) => class extends superclass {
+      // only check estimatedFee when isSendAll=false
+      if (!isSendAll) {
 
-      try {
-        ({ fee: estimatedFee } = await getChainFeeEstimate({ lnd, send_to: sendTo }))
-      } catch (err) {
-        const error = `Unable to estimate fee for on-chain transaction`
-        onchainLogger.error({ err, sendTo, success: false }, error)
-        throw new LoggedError(error)
-      }
+        let estimatedFee
 
-      // case where there is not enough money available within lnd on-chain wallet
-      if (onChainBalance < amount + estimatedFee) {
-        // TODO: add a page to initiate the rebalancing quickly
-        throw new RebalanceNeededError(undefined, {logger: onchainLogger, onChainBalance, amount, estimatedFee, sendTo, success: false})
-      }
+        const sendTo = [{ address, tokens: amount }]
+
+        try {
+          ({ fee: estimatedFee } = await getChainFeeEstimate({ lnd, send_to: sendTo }))
+        } catch (err) {
+          const error = `Unable to estimate fee for on-chain transaction`
+          onchainLogger.error({ err, sendTo, success: false }, error)
+          throw new LoggedError(error)
+        }
+  
+        // case where there is not enough money available within lnd on-chain wallet
+        if (onChainBalance < amount + estimatedFee) {
+          // TODO: add a page to initiate the rebalancing quickly
+          throw new RebalanceNeededError(undefined, {logger: onchainLogger, onChainBalance, amount, estimatedFee, sendTo, success: false})
+        }
 
-      //add a flat fee on top of onchain miner fees
-      estimatedFee += this.user.withdrawFee
+        //add a flat fee on top of onchain miner fees
+        estimatedFee += this.user.withdrawFee
 
-      // case where the user doesn't have enough money
-      if (balance.total_in_BTC < amount + estimatedFee) {
-        throw new InsufficientBalanceError(undefined, {logger: onchainLogger})
+        // case where the user doesn't have enough money
+        if (balance.total_in_BTC < amount + estimatedFee) {
+          throw new InsufficientBalanceError(undefined, { logger: onchainLogger })
+        }
+
+        amountToSend = amount
+      }
+      // else when isSendAll, fees will need to be deducted from the amount
+      else {
+
+        /// TODO? is this check needed?
+        // case where there is not enough money available within lnd on-chain wallet
+        if (onChainBalance < amount) {
+          // TODO: add a page to initiate the rebalancing quickly
+          throw new RebalanceNeededError(undefined, {logger: onchainLogger, amount, success: false})
+        }
+
+        amountToSend = amount - this.user.withdrawFee
       }
 
 
       return lockExtendOrThrow({lock, logger: onchainLogger}, async () => {
 
-        try {
-          ({ id } = await sendToChainAddress({ address, lnd, tokens: amount }))
+        try {          
+          ({ id } = await sendToChainAddress({ address, is_send_all: isSendAll, lnd, tokens: amountToSend }))
```

Comment:
nope: `is_send_all: isSendAll`. we're sending the sendAll from the perspective of the user. here you would send everything within the hot wallet.